R-package/R/xgb.Booster.R

```diff
@@ -275,25 +280,22 @@ predict.xgb.Booster <- function(object, newdata, missing = NA, outputmargin = FA
     )
   }
 
-  if (NVL(xgb.booster_type(object), '') == 'gblinear' || is.null(ntreelimit))
-    ntreelimit <- 0
 
-  if (ntreelimit != 0 && is.null(iterationrange)) {
-    ## only ntreelimit, initialize iteration range
-    iterationrange <- c(0, 0)
-  } else if (ntreelimit == 0 && !is.null(iterationrange)) {
-    ## only iteration range, handle 1-based indexing
-    iterationrange <- c(iterationrange[1] - 1, iterationrange[2] - 1)
-  } else if (ntreelimit != 0 && !is.null(iterationrange)) {
-    ## both are specified, let libgxgboost throw an error
+  if (!is.null(iterationrange)) {
+    if (is.character(iterationrange) &&
+        length(iterationrange) == 1 &&
+        iterationrange == "all") {
+      iterationrange <- c(0, 0)
+    } else {
```

Comment:
since the only valid value is "all", maybe we can check:
``` R
if (is.character(iterationrange)) {
    stopifnot(iterationrange == "all")
    iterationrange <- c(0, 0)
} else {
    ...
}
```

Reply:
Modified.